File: editorial/chap6-16_20/prob6-18.java
import java.util.*;

class Main {
	public static void main(String[] args) {
		// 入力
		Scanner sc = new Scanner(System.in);
		long A = sc.nextLong();
		long B = sc.nextLong();
		
		// 最小公倍数が 10^18 を超えるか判定
		if (A / GCD(A, B) > 1000000000000000000L / B) {
			System.out.println("Large");
		}
		else {
			System.out.println(A / GCD(A, B) * B);
		}
	}
	static long GCD(long A, long B) {
		if (B == 0) return A;
		return GCD(B, A % B);
	}
}

File: editorial/chap6-16_20/prob6-18.c
#include <stdio.h>

long long GCD(long long A, long long B) {
	if (B == 0) return A;
	return GCD(B, A % B);
}

int main() {
	// 入力
	long long A, B;
	scanf("%lld%lld", &A, &B);

	// 最小公倍数が 10^18 を超えるかどうか判定
	if (A / GCD(A, B) > 1000000000000000000 / B) {
		printf("Large\n");
	}
	else {
		printf("%lld\n", A / GCD(A, B) * B);
	}
	return 0;
}